contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,517 @@
+type CStorPoolClusterCondition struct {
+	// Type of CSPC condition.
+	Type CSPCConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=DeploymentConditionType"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,6,opt,name=lastUpdateTime"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,7,opt,name=lastTransitionTime"`
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+
+// CStorPoolClusterList is a list of CStorPoolCluster resources
+type CStorPoolClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorPoolCluster `json:"items"`
+}
+
+```
+
+### CSPI Schema
+
+The CSPI resuses the CSPC schema and following is the current schema in go struct.
+
+```go
+// CStorPoolInstance describes a cstor pool instance resource created as custom resource.
+type CStorPoolInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec           CStorPoolInstanceSpec `json:"spec"`
+	Status         CStorPoolStatus       `json:"status"`
+	VersionDetails VersionDetails        `json:"versionDetails"`
+}
+
+// CStorPoolInstanceSpec is the spec listing fields for a CStorPoolInstance resource.
+type CStorPoolInstanceSpec struct {
+	// HostName is the name of kubernetes node where the pool
+	// should be created. This is filled by CSPC-operator.
+	HostName string `json:"hostName"`
+	// NodeSelector is the labels that is filled by CSPC-operator, only
+	// for informational purpose.
+	NodeSelector map[string]string `json:"nodeSelector"`
+	// PoolConfig is the default pool config that applies to the
+	// pool on node.
+	PoolConfig PoolConfig `json:"poolConfig"`
+	// RaidGroups is the group containing block devices
+	RaidGroups []RaidGroup `json:"raidGroup"`
+}
+
+// CStorPoolStatus is for handling status of pool.
+type CStorPoolStatus struct {
+	Phase    CStorPoolPhase        `json:"phase"`
+	Capacity CStorPoolCapacityAttr `json:"capacity"`
+	// LastTransitionTime refers to the time when the phase changes
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	LastUpdateTime     metav1.Time `json:"lastUpdateTime,omitempty"`
+	Message            string      `json:"message,omitempty"`
+}
+
+// CStorPoolCapacityAttr stores the pool capacity related attributes.
+type CStorPoolCapacityAttr struct {
```

Comment:
`CStorPoolCapacityDetails` as we are having `CStorVolumeCapacityDetails in case of volumes ?